src/main/java/org/broadinstitute/hellbender/tools/walkers/varianteval/evaluators/VariantEvaluator.java

```diff
@@ -1,41 +1,42 @@
 package org.broadinstitute.hellbender.tools.walkers.varianteval.evaluators;
 
 import htsjdk.variant.variantcontext.VariantContext;
-import org.broadinstitute.hellbender.engine.FeatureContext;
-import org.broadinstitute.hellbender.engine.ReadsContext;
-import org.broadinstitute.hellbender.engine.ReferenceContext;
 import org.broadinstitute.hellbender.exceptions.GATKException;
-import org.broadinstitute.hellbender.tools.walkers.varianteval.VariantEval;
+import org.broadinstitute.hellbender.tools.walkers.varianteval.VariantEvalEngine;
+import org.broadinstitute.hellbender.tools.walkers.varianteval.util.VariantEvalContext;
 
 public abstract class VariantEvaluator implements Comparable<VariantEvaluator> {
-    private VariantEval walker;
     private final String simpleName;
+    private final VariantEvalEngine engine;
 
-    protected VariantEvaluator(String simpleName) {
-        this.simpleName = simpleName;
+    public VariantEvaluator(VariantEvalEngine engine) {
+        this.engine = engine;
+        this.simpleName = getClass().getSimpleName();
     }
 
-    protected VariantEvaluator() {
-        this.simpleName = getClass().getSimpleName();
+    //Note: this is used by DISCVR-seq / VariantQC
+    public VariantEvaluator(VariantEvalEngine engine, final String simpleName) {
+        this.engine = engine;
+        this.simpleName = simpleName;
     }
 
-    public void initialize(VariantEval walker) {
-        this.walker = walker;
+    public void initialize() {
+
```

Comment:
I thought passing the engine to constructors would eliminate the need for this initialize method. Can it be removed ?

Reply:
VariantSummary is an example of one class using this to initialize variables. It doesnt seem crazy to me to keep this? In theory this class could store some internal "hasInitialized" boolean, and lazily do this work in the update() method, but initialize() doesnt seem that harmful to exist? Example below:

```
    @Override
    public void initialize() {
        this.knownCNVs = getEngine().getVariantEvalArgs().getKnownCNVsFile();

        nSamples = getEngine().getSampleNamesForEvaluation().size();
        countsPerSample = new TypeSampleMap(getEngine().getSampleNamesForEvaluation());
        transitionsPerSample = new TypeSampleMap(getEngine().getSampleNamesForEvaluation());
        transversionsPerSample = new TypeSampleMap(getEngine().getSampleNamesForEvaluation());
        allVariantCounts = new TypeSampleMap(getEngine().getSampleNamesForEvaluation());
        knownVariantCounts = new TypeSampleMap(getEngine().getSampleNamesForEvaluation());
        depthPerSample = new TypeSampleMap(getEngine().getSampleNamesForEvaluation());
    }

```

Reply:
I thought about this and I agree - initialize() doesnt really need to exist. I moved the two classes that were actually using it to do that work in their constructors.